src/main/java/me/ramidzkh/fabrishot/event/FramebufferCaptureCallback.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2021 Ramid Khan
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package me.ramidzkh.fabrishot.event;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Callback for right after a framebuffer is captured, but before it is saved
+ */
+public interface FramebufferCaptureCallback {
+
+    Event<FramebufferCaptureCallback> EVENT = EventFactory.createArrayBacked(FramebufferCaptureCallback.class, callbacks -> buffer -> {
+        for (FramebufferCaptureCallback callback : callbacks) {
+            callback.onCapture(buffer);
```

Comment:
I need the dimensions of the image here, should be easy to pass that through from `FramebufferCapturer`. Would also be good to specify the format is RGB (3 components) in the javadoc.